packages/contract/src/contract-factory.ts

```diff
@@ -235,28 +408,64 @@ export default class ContractFactory {
 
     const { contractId, transactionRequest } = this.createTransactionRequest(deployOptions);
 
-    const account = this.getAccount();
-
-    const txCost = await account.getTransactionCost(transactionRequest);
-
-    const { maxFee: setMaxFee } = deployOptions;
-
-    if (isDefined(setMaxFee)) {
-      if (txCost.maxFee.gt(setMaxFee)) {
-        throw new FuelError(
-          ErrorCode.MAX_FEE_TOO_LOW,
-          `Max fee '${deployOptions.maxFee}' is lower than the required: '${txCost.maxFee}'.`
-        );
-      }
-    } else {
-      transactionRequest.maxFee = txCost.maxFee;
-    }
-
-    await account.fund(transactionRequest, txCost);
+    await this.fundTransactionRequest(transactionRequest, deployOptions);
 
     return {
       contractId,
       transactionRequest,
     };
   }
+
+  /**
+   * Create a blob transaction request, used for deploying contract chunks.
+   *
+   * @param options - options for creating a blob transaction request.
+   * @returns a populated BlobTransactionRequest.
+   */
+  private blobTransactionRequest(options: { bytecode: BytesLike } & DeployContractOptions) {
+    const { bytecode } = options;
+    return new BlobTransactionRequest({
+      blobId: hash(bytecode),
+      witnessIndex: 0,
+      witnesses: [bytecode],
+      ...options,
+    });
+  }
+
+  /**
+   * Get the maximum chunk size for deploying a contract by chunks.
+   */
+  private getMaxChunkSize(
+    deployOptions: DeployContractOptions,
+    chunkSizeOverride: number = CHUNK_SIZE_OVERRIDE
```

Comment:
I'm not entirely convinced on the naming of this variable as override gives the impression that we're overriding an entire value. However, we're using this override to calculate an offset multiplier for the contract size threshold.

`CHUNK_SIZE_THRESHOLD` - this would be defaulted to `95%`, so in essence, the result of `override`:
```ts
const override = CHUNK_SIZE_THRESHOLD = (1 - chunkSizeOverride);
```

I am not entirely convinced with this naming either, but I'd do away with the inverse option (`1 - chunkSizeOverride`).

Reply:
Removed the inverse and changed to `chunkSizeMultiplier`.